os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -456,14 +443,9 @@ static void arm_assert(void)
 
 static inline void print_assert_detail(const uint8_t *filename, int lineno, struct tcb_s *fault_tcb, uint32_t asserted_location)
 {
-  board_autoled_on(LED_ASSERTION);
-
-#ifdef CONFIG_SYSTEM_REBOOT_REASON
-	reboot_reason_write_user_intended();
-#endif
-
+	lldbg_noarg("===========================================================Assertion_Details===========================================\n");
 #if CONFIG_TASK_NAME_SIZE > 0
```

Comment:
To make full sentence, you used two ways.
1. smp
2. task name

The SMP splits the line and adds the value.
But the task name just repeats twice using `#ifdef ~ #else`.

Let's use the same way for this.

Reply:
I suggest using just single format. I mean, even we can't print a task name ans a num of cpu, we can keep the same log format and print `N/A` without the config enabled.
For example
```
#ifdef CONFIG_SMP
#define LOG_CPU_INDEX   up_cpu_index()
#else
#define LOG_CPU_INDEX   (0)
#endif

#if CONFIG_TASK_NAME_SIZE > 0
#define LOG_TASK_NAME  fault_tcb->name
#else
#define LOG_TASK_NAME   "N/A"
#endif

printf("Assertion Failed CPU%d at file: %s line %d task: $s pid: %d\n",
           LOG_CPU_INDEX, filename, lineno, LOG_TASK_NAME, fault_tcb->pid);
```

Current code is too complex.

Reply:
I have pushed the changes. Let me know if it need any further change.